src/app/vitessce-grid-layout/VitessceGridLayout.js

```diff
@@ -44,36 +45,43 @@ export default function VitessceGridLayout(props) {
   const style = (
     <style>
       {`
-          ${draggableHandle} {
-            cursor: grab;
-          }
-          ${draggableHandle}:active {
-            cursor: grabbing;
-          }
+        ${draggableHandle} {
+          cursor: grab;
+        }
+        ${draggableHandle}:active {
+          cursor: grabbing;
+        }
      `}
     </style>
   );
 
-  useEffect(() => {
-    if (readyComponentKeys.size === Object.keys(gridComponents).length) {
-      // The sets are now equal.
-      onAllReady();
+  const onLayoutChange = (newLayout) => {
+    if (newLayout.length === Object.entries(gridComponents).length) {
+      newLayout.forEach((nextC) => {
+        const id = nextC.i;
+        const prevC = gridComponents[id];
+        const i = parseInt(id.substring(id.indexOf(COMPONENT_ID_PREFIX) + 1), 10);
+        const nextProps = {
+          x: nextC.x, y: nextC.y, w: nextC.w, h: nextC.h,
+        };
+        const prevProps = {
+          x: prevC.x, y: prevC.y, w: prevC.w, h: prevC.h,
+        };
+        if (!isEqual(nextProps, prevProps)) {
+          onComponentChange(i, nextProps);
+        }
+      });
     }
-  }, [readyComponentKeys, gridComponents, onAllReady]);
+  };
```

Comment:
Always a little bit wary of passing in these `const` functions without `useCallback` into components....sometimes it seems like it can go poorly but I have never been super clear on this.

Reply:
I think it just depends on whether there is a child component that stores a reference to the function. I think there can be an infinite loop if the child component checks the reference and calls the callback during its initialization. It does not look like there is an issue like that happening here. If we use useCallback here it would depend on `gridComponents` which is computed on every render anyway